qcodes/instrument_drivers/stanford_research/SR830.py

```diff
@@ -235,11 +334,23 @@ def parse_offset_get(s):
                                'GPIB': '1\n',
                            })
 
-        # Auto functions
-        self.add_function('auto_gain', call_cmd='AGAN')
-        self.add_function('auto_reserve', call_cmd='ARSV')
-        self.add_function('auto_phase', call_cmd='APHS')
-        self.add_function('auto_offset', call_cmd='AOFF {0}', args=[Enum(1, 2, 3)])
+        # Channel setup
+        for ch in range(1, 3):
+
+            # detailed validation and mapping performed in set/get functions
+            self.add_parameter('ch{}_ratio'.format(ch),
+                               label='Channel {} ratio'.format(ch),
+                               get_cmd=partial(self._get_ch_ratio, ch),
+                               set_cmd=partial(self._set_ch_ratio, ch),
+                               vals=Strings())
+            self.add_parameter('ch{}_display'.format(ch),
+                               label='Channel {} display'.format(ch),
+                               get_cmd=partial(self._get_ch_display, ch),
+                               set_cmd=partial(self._set_ch_display, ch),
+                               vals=Strings())
+            self.add_parameter('ch{}_databuffer'.format(ch),
+                               channel=ch,
+                               parameter_class=ChannelBuffer)
 
         # Data transfer
         self.add_parameter('X',
```

Comment:
Yes. The internal buffer is filled simultaneously for both channels, which is why there is only one `buffer_SR` etc., but polling is done from either of the two (but not both).